MinecraftClient/Protocol/Handlers/DataTypes.cs

```diff
@@ -610,22 +610,27 @@ private object ReadNbtField(Queue<byte> cache, int fieldType)
                         d.Add(ReadNextVarInt(cache));
                         d.Add(ReadNextVarInt(cache));
                         d.Add(ReadNextVarInt(cache));
-                        Value = d;
+                        value = d;
                         break;
                     case 17: // Optional VarInt
                         if (ReadNextBool(cache))
                         {
-                            Value = ReadNextVarInt(cache);
+                            value = ReadNextVarInt(cache);
                         }
                         break;
                     case 18: // Pose
-                        Value = ReadNextVarInt(cache);
+                        value = ReadNextVarInt(cache);
                         break;
                     default:
-                        throw new System.IO.InvalidDataException("Unknown Metadata Type ID " + Type + ". Is this up to date for new MC Version?");
+                        throw new System.IO.InvalidDataException("Unknown Metadata Type ID " + type + ". Is this up to date for new MC Version?");
                 }
-                data.Add(Key, Value);
-                Key = ReadNextByte(cache);
+                // issue #1209: An item with the same index key has been added
+                // That shouldn't happen, but if yes then we over-write the existing one
+                if (data.ContainsKey(key))
+                    data[key] = value;
+                else
+                    data.Add(key, value);
```

Comment:
You can use `data[key] = value` directly even if the entry is not yet in the dictionary. No need to check before setting the value 😉